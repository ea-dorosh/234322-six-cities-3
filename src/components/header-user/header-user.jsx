import React, {PureComponent} from "react";
// import PropTypes from "prop-types";
import {connect} from "react-redux";
import {AuthorizationStatus} from "../../reducer/user/user";
import {Link} from "react-router-dom";
import {getAuthorizationStatus} from "../../reducer/user/selectors";


class HeaderUser extends PureComponent {
  constructor(props) {
    super(props);

  }

  render() {

    // eslint-disable-next-line react/prop-types
    const {authorizationStatus, userProperties} = this.props;

    return (
      <ul className="header__nav-list">
        <li className="header__nav-item user">
          {authorizationStatus === AuthorizationStatus.NO_AUTH ?
            <Link
              to={`/dev-auth`}
              className="header__nav-link header__nav-link--profile" href="#">
              <div className="header__avatar-wrapper user__avatar-wrapper">
              </div>
              <span className="header__user-name user__name">Sign in</span>
            </Link>
            :
            <a
              className="header__nav-link header__nav-link--profile" href="#">
              <div className="header__avatar-wrapper user__avatar-wrapper">
              </div>
              <span className="header__user-name user__name">
                {/* eslint-disable-next-line react/prop-types */}
                {userProperties.email}
              </span>
            </a>
          }
        </li>
      </ul>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    authorizationStatus: getAuthorizationStatus(state),
    userProperties: state.USER.userProperties,
  };
};

export default connect(mapStateToProps)(HeaderUser);

