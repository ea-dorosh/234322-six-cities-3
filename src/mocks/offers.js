import {getRandomNumber, getRandomElement} from "../utils";

const PRICE = {
  MIN: 50,
  MAX: 500
};

const offersNames = [
  `Beautiful & luxurious apartment at great location`,
  `Nice, cozy, wood and stone place`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`
];

const holderNames = [
  `Angelina`,
  `Max`,
];

const offersTypes = [
  `apartment`,
  `room`,
  `house`,
  `hotel`
];

const offersCords = [
  [52.3909553943508, 4.85309666406198],
  [52.369553943508, 4.85309666406198],
  [52.3909553943508, 4.929309666406198],
  [52.3809553943508, 4.939309666406198]
];

const createOffer = (index) => {
  const bedRoomQuantity = getRandomNumber(1, 6);
  const maxGuestQuantity = bedRoomQuantity * 2;
  const holderName = getRandomElement(holderNames);

  return {
    img: `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
    photos: [
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
      `img/apartment-0${getRandomNumber(1, 3)}.jpg`,
    ],
    isPremium: !!(index % 2),
    price: getRandomNumber(PRICE.MIN, PRICE.MAX),
    name: offersNames[index],
    type: getRandomElement(offersTypes),
    rating: parseFloat(`${getRandomNumber(2, 4)}.${getRandomNumber(0, 9)}`),
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedRoomQuantity,
    maxGuestQuantity,
    details: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    holder: {
      holderName,
      img: `img/avatar-${holderName.toLowerCase()}.jpg`,
      isSuper: !(index % 2),
    },
    coords: {
      x: offersCords[index][0],
      y: offersCords[index][1],
    }
  };
};

const createOffers = (amount) => {
  const offers = [];

  for (let i = 0; i < amount; i++) {
    const offer = createOffer(i);

    offers.push(offer);
  }
  return offers;

};


export {createOffers};
