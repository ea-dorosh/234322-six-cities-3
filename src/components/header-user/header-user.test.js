import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {HeaderUser} from "./header-user.jsx";
import {userPropertiesMock} from "../../mocks/offers_for_test";

it(`Should HeaderUser render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter>
          <HeaderUser
            authorizationStatus={`string`}
            userProperties={userPropertiesMock}
          />
        </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
