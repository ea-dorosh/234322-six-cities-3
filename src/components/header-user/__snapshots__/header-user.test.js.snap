// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should HeaderUser render correctly 1`] = `
<ul
  className="header__nav-list"
>
  <li
    className="header__nav-item user"
  >
    <a
      className="header__nav-link header__nav-link--profile"
      href="/favorites"
      onClick={[Function]}
    >
      <div
        className="header__avatar-wrapper user__avatar-wrapper"
      />
      <span
        className="header__user-name user__name"
      >
        ea-dorosh@yandex.ru
      </span>
    </a>
  </li>
</ul>
`;
